src/test/java/gim/testutil/TypicalExercises.java

```diff
@@ -75,4 +79,94 @@ public static ExerciseTracker getTypicalExerciseTracker() {
     public static List<Exercise> getTypicalExercises() {
         return new ArrayList<>(Arrays.asList(ALICE, BENSON, CARL, DANIEL, ELLE, FIONA, GEORGE));
     }
+
+    /**
+     * Returns an {@code ExerciseTracker} with unsorted same exercises but different dates.
+     */
+    public static ExerciseTracker getUnsortedSameExercisesDifferentDatesExerciseTracker() {
+        ExerciseTracker ab = new ExerciseTracker();
+        for (Exercise exercise : getUnsortedSameExercisesDifferentDates()) {
+            ab.addExercise(exercise);
+        }
+        return ab;
+    }
+
+    public static List<Exercise> getUnsortedSameExercisesDifferentDates() {
+        return new ArrayList<>(Arrays.asList(ARM_CURLS_2, ARM_CURLS));
+    }
+
+    /**
+     * Returns an {@code ExerciseTracker} with sorted same exercises but different dates.
+     */
+    public static ExerciseTracker getSortedSameExercisesDifferentDatesExerciseTracker() {
+        ExerciseTracker ab = new ExerciseTracker();
+        for (Exercise exercise : getSortedSameExercisesDifferentDates()) {
+            ab.addExercise(exercise);
+        }
+        return ab;
+    }
+
+    public static List<Exercise> getSortedSameExercisesDifferentDates() {
+        return new ArrayList<>(Arrays.asList(ARM_CURLS, ARM_CURLS_2));
+    }
+
+    /**
+     * Returns an {@code ExerciseTracker} with unsorted different exercises but same dates.
+     */
+    public static ExerciseTracker getUnsortedDifferentExercisesSameDatesExerciseTracker() {
+        ExerciseTracker ab = new ExerciseTracker();
+        for (Exercise exercise : getUnsortedDifferentExercisesSameDates()) {
+            ab.addExercise(exercise);
+        }
+        return ab;
+    }
+
+    public static List<Exercise> getUnsortedDifferentExercisesSameDates() {
+        return new ArrayList<>(Arrays.asList(BENCH_PRESS, ARM_CURLS));
+    }
+
+    /**
+     * Returns an {@code ExerciseTracker} with sorted different exercises but same dates.
+     */
+    public static ExerciseTracker getSortedDifferentExercisesSameDatesExerciseTracker() {
+        ExerciseTracker ab = new ExerciseTracker();
+        for (Exercise exercise : getSortedDifferentExercisesSameDates()) {
+            ab.addExercise(exercise);
+        }
+        return ab;
+    }
+
+    public static List<Exercise> getSortedDifferentExercisesSameDates() {
+        return new ArrayList<>(Arrays.asList(ARM_CURLS, BENCH_PRESS));
+    }
+
+    /**
+     * Returns an {@code ExerciseTracker} with unsorted different exercises and different dates.
+     */
+    public static ExerciseTracker getUnsortedDifferentExercisesDifferentDatesExerciseTracker() {
+        ExerciseTracker ab = new ExerciseTracker();
```

Comment:
The naming of variable `ab` here is a little confusing.

Reply:
I do agree with this, will make the necessary changes!